src/classes/exec-sync-utils.ts

```diff
@@ -1,15 +1,26 @@
 import { execSync, ExecSyncOptionsWithBufferEncoding } from "child_process";
 
 /**
- * runs the given command in a shell, vulnerable to injection
+ * runs the given command in a shell, denies attempts of injection
  * 
  * @param {string} command command to be run in shell
  * @param {ExecSyncOptionsWithBufferEncoding} options options to be provided to the shell
  * @returns {Promise<string | void>} if command outputs, will provide it as string
  */
 export const runExecSync = async (command: string, options?: ExecSyncOptionsWithBufferEncoding): Promise<string | void> => {
   try {
-    return execSync(command, options)?.toString(); //NOSONAR
+    /**
+     * Deny executable, shell, installation, registry and python files,
+     * files that are run in a sandbox without checks as well as files with possible malware included
+     */
+    if (command.search(/;|\.exe$|\.com$|\.dll$|\.vbs$|\.vb$|\.vbe$|\.pif$|\.application$|\.gadget$|\.ws$|\.wsf$|\.reg$/) == -1) {
+      if (command.search(/\.msi$|\.msp$|\.scr$|\.hta$|\.cpl$|\.msc$|\.jar$|\.bat$|\.cmd$|\.js$|\.jse$|\.wsc$|\.wsh$/) == -1) {
+        if (command.search(/\.ps.$|\.ps.xml$|\.psc.$|\.msh$|\.msh.$|\.mshxml$|\.msh.xml$|\.pdf$|\.sfx$|\.tmp$|\.py$/) == -1) {
```

Comment:
Should probably stop execution of command this was called in. Not executing 1 or more execSyncs will cause problems if any are to be driven after.

Also these would be better as variables, we might find more to add to this list of things to block.

Reply:
File extensions have now been added into an array of regular expressions. Added errors.